packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -1737,7 +1738,10 @@ export class ProjectModel {
                                               m.id === d.dashboard_version_id,
                                       )?.newId!,
                                   tab_uuid: dashboardTabsMapping.find(
-                                      (m) => m.uuid === d.tab_uuid,
+                                      (m) =>
+                                          m.uuid === d.tab_uuid &&
+                                          m.newDashboardVersionId ===
+                                              d.dashboard_version_id,
```

Comment:
@Shifan-Gu from what I am seeing while testing, this will return undefined since `newDashboardVersionId` in the tabs mapping never matches the tile's `dashboard_version_id` and therefore the value of `tab_uuid` in the preview project `dashboard_tile` rows will always be `NULL`

I believe we can go around this by adding `dashboardVersionId: dashboardTabs[i].dashboard_version_id` to the tabs mapping and use it in this check instead of `m.newDashboardVersionId`.
After this we can probably remove `newDashboardVersionId` from the tabs mapping as it won't be used I believe.

Reply:
Code looks good, thank you @Shifan-Gu 🚀